src/decorators.ts

```diff
@@ -48,4 +49,24 @@ export function Exclude(options?: ExcludeOptions) {
         const metadata = new ExcludeMetadata(object instanceof Function ? object : object.constructor, propertyName, options || {});
         defaultMetadataStorage.addExcludeMetadata(metadata);
     };
+}
+
+/**
+ * Return the object with the exposed properties only.
+ */
+export function JsonView(params?: {}, method?: string): Function {
+
+    return function (target: Function, propertyKey: string, descriptor: PropertyDescriptor) {
+        const classTransformer: any = new ClassTransformer();
+        const originalMethod = descriptor.value;
+
+        descriptor.value = function(...args: any[]) {
+            let transformer: Function = typeof method === "string" && method ? classTransformer[method] : classTransformer.classToPlain;
+            let result: any = originalMethod.apply(this, args);
```

Comment:
once method will be a string literal type you can use simple switch-cases to make this code both more readable and understandable + make it more typesafe.